lib/elixir/lib/calendar/datetime.ex

```diff
@@ -930,11 +931,31 @@ defmodule DateTime do
       iex> DateTime.to_iso8601(dt, :basic)
       "20000229T230007-0400"
 
+      iex> dt = %DateTime{year: 2000, month: 2, day: 29, zone_abbr: "AMT",
+      ...>                hour: 23, minute: 0, second: 7, microsecond: {0, 0},
+      ...>                utc_offset: -14400, std_offset: 0, time_zone: "America/Manaus"}
+      iex> DateTime.to_iso8601(dt, :extended, 3600)
+      "2000-03-01T04:00:07+01:00"
+
+      iex> dt = %DateTime{year: 2000, month: 2, day: 29, zone_abbr: "AMT",
+      ...>                hour: 23, minute: 0, second: 7, microsecond: {0, 0},
+      ...>                utc_offset: -14400, std_offset: 0, time_zone: "America/Manaus"}
+      iex> DateTime.to_iso8601(dt, :extended, 0)
+      "2000-03-01T03:00:07+00:00"
+
+      iex> dt = %DateTime{year: 2000, month: 3, day: 01, zone_abbr: "UTC",
+      ...>                hour: 03, minute: 0, second: 7, microsecond: {0, 0},
+      ...>                utc_offset: 0, std_offset: 0, time_zone: "Etc/UTC"}
+      iex> DateTime.to_iso8601(dt, :extended, 0)
+      "2000-03-01T03:00:07Z"
+
+      iex> {:ok, dt, offset} = DateTime.from_iso8601("2000-03-01T03:00:07Z")
+      iex> "2000-03-01T03:00:07Z" = DateTime.to_iso8601(dt, :extended, offset)
   """
-  @spec to_iso8601(Calendar.datetime(), :extended | :basic) :: String.t()
-  def to_iso8601(datetime, format \\ :extended)
+  @spec to_iso8601(Calendar.datetime(), :basic | :extended, nil | integer()) :: String.t()
+  def to_iso8601(datetime, format \\ :extended, offset \\ nil)
```

Comment:
Can the default be zero?

Reply:
It's not zero so the function can tell between "override using this offset" and "use the offset for the timezone supplied".